src/Dapr.PluggableComponents.AspNetCore/WebApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,52 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Dapr.PluggableComponents;
+
+public static class WebApplicationBuilderExtensions
+{
+    public static WebApplicationBuilder AddDaprPluggableComponentsServices(this WebApplicationBuilder builder, DaprPluggableComponentsApplicationOptions options)
+    {
+        string socketExtension = options.SocketExtension
+            ?? Environment.GetEnvironmentVariable(Constants.EnvironmentVariables.DaprComponentsSocketsExtension)
+            ?? Constants.Defaults.DaprComponentsSocketsExtension;
+
+        // TODO: Add support for native (i.e. non-WSL) Windows.
+        string socketFolder = options.SocketFolder
+            ?? Environment.GetEnvironmentVariable(Constants.EnvironmentVariables.DaprComponentsSocketsFolder)
+            ?? Constants.Defaults.DaprComponentsSocketsFolder;
+
+        string socketName = options.SocketName
+            ?? Environment.GetEnvironmentVariable(Constants.EnvironmentVariables.DaprComponentsSocketsName)
+            ?? throw new ArgumentException("The socket name was not specified via the options or environment variable.", nameof(options));
+
+        string socketPath = Path.Join(socketFolder, socketName + socketExtension);
+
+        Directory.CreateDirectory(socketFolder);
```

Comment:
The only thing that is missing is to setup the socket permissions [see this](https://github.com/artursouza/kubecon-na-2022/blob/main/pluggable-components/memstore/Program.cs#L62)

Reply:
Yes, I'm just getting to testing within k8s itself, where this is necessary.

Reply:
`docker-compose` is another use-case that might require that -- we faced this problem when playing with `docker-compose` actually ;)

Reply:
Permissions are now being updated (tested in k8s).  As an aside, the Dapr sidecar seems to get stuck when the permissions are not correct, rather than generating any sort of error.  I'm not sure if the sidecar can identify that situation in order to provide an error.